core/server/master/src/main/java/alluxio/master/meta/PathProperties.java

```diff
@@ -146,10 +146,12 @@ public Journaled getDelegate() {
      */
     public void addProperties(Supplier<JournalContext> ctx, String path,
         Map<PropertyKey, String> properties) {
-      Map<String, String> newProperties = mProperties.getOrDefault(path, new HashMap<>());
-      properties.forEach((key, value) -> newProperties.put(key.getName(), value));
-      applyAndJournal(ctx, PathPropertiesEntry.newBuilder().setPath(path)
-          .putAllProperties(newProperties).build());
+      if (!properties.isEmpty()) {
```

Comment:
can we move `addProperties` and `removeProperties` outside of `State`? The correctness of `State` is critical for the journal to be able to replay correctly, so I'd like to keep it as simple as possible. For example, we create journal entries in `InodeTree` and then apply them to `InodeTreePersistentState` instead of creating the entries in `InodeTreePersistentState`.

In general, all methods in `State` should take a journal entries as arguments